src/wad/tex.rs

```diff
@@ -143,18 +154,18 @@ impl TextureDirectory {
     }
 
 
-    pub fn build_texture_atlas<'a, T: IntoIterator<Item = &'a WadName>>(
-            &'a self, names_iter: T) -> (TransparentImage, BoundsLookup) {
-        let entries = ordered_atlas_entries(&self.animated_walls,
-                                            |n| self.texture(&n),
-                                            names_iter);
+    pub fn build_texture_atlas<'a, T: IntoIterator<Item = &'a WadName>>
```

Comment:
This is terrible! Could you switch to a where clause here to remove the madness?